dbux-data/src/indexes/CollectionIndex.js

```diff
@@ -16,54 +95,67 @@ export default class CollectionIndex {
    */
   _byKey;
 
-  constructor(collectionName, indexName, { addOnNewData = true, stringKey = false } = EmptyObject) {
+  constructor(collectionName, indexName, { addOnNewData = true, isMap = false, isContainerSet = false } = EmptyObject) {
     this.collectionName = collectionName;
     this.name = indexName;
     this.log = newLogger(`${indexName} (Index)`);
     this.addOnNewData = addOnNewData;
-    this.stringKey = stringKey;
-  }
+    this.isMap = isMap;
+    this.isContainerSet = isContainerSet;
 
-  _init(dp) {
-    this.dp = dp;
-    if (this.stringKey) {
-      this._byKey = {};
+    if (this.isMap) {
+      this._manager = MapIndexManager;
     }
     else {
-      this._byKey = [];
+      this._manager = ArrayIndexManager;
+    }
+    this._manager = Object.fromEntries(
+      Object.entries(this._manager).map(([name, func]) => [name, func.bind(this)])
+    );
+
+    if (this.isContainerSet) {
+      this._containerMethods = SetContainerMethods;
     }
+    else {
+      this._containerMethods = ArrayContainerMethods;
+    }
+    this._containerMethods = Object.fromEntries(
+      Object.entries(this._containerMethods).map(([name, func]) => [name, func.bind(this)])
```

Comment:
Spread chained function calls over multiple lines for readability. `\n` before `map`.